utils/api/helpers/requestHandler.ts

```diff
@@ -273,6 +271,9 @@ export class RequestHandler {
       // Save the result to the client's partnerBenefitStatus field, which is used TBD
       input.client.partnerBenefitStatus.gisResultEligibility =
         partnerGis.eligibility
+      // Save the result to the partner's partnerBenefitStatus field, which is used for partner's alw eligibility
+      input.partner.partnerBenefitStatus.gisResultEligibility =
+        clientGis.eligibility
     }
```

Comment:
Was this the bug?

Reply:
haha yeah, it was a tricky one. the issue was that the client's GIS entitlement depends on the partner's ALW eligibility, which depends on the client's GIS eligibility, (which depends on the client's OAS eligibility) ...... and I was not saving the client's GIS eligibility into the object that determines the partner's ALW eligibility. It's a huge mouthful and it took a while to finally understand it, hence why I put a ton of comments explaining why I'm doing each update!